engine/api/services.go

```diff
@@ -51,96 +54,88 @@ func (api *API) getServiceHandler() service.Handler {
-		defer tx.Rollback() // nolint
+	if exists && srv.Type != data.Type {
+		return sdk.WrapError(sdk.ErrForbidden, "cannot register service %s of type %s for consumer %s while existing service type is different", data.Name, data.Type, consumer.ID)
+	}
 
-		// Try to find the service, and keep; else generate a new one
-		srv, err := services.LoadByConsumerID(ctx, tx, consumer.ID)
-		if err != nil && !sdk.ErrorIs(err, sdk.ErrNotFound) {
+	// Update or create the service
+	session := getAuthSession(ctx)
+	if session == nil {
+		return sdk.NewErrorFrom(sdk.ErrUnauthorized, "missing registered session")
+	}
+	if exists {
+		srv.Update(*data)
+		if err := services.Update(ctx, tx, srv); err != nil {
 			return err
 		}
-		exists := srv != nil
-
-		if exists && srv.Type != data.Type {
-			return sdk.WrapError(sdk.ErrForbidden, "cannot register service %s of type %s for consumer %s while existing service type is different", data.Name, data.Type, consumer.ID)
-		}
-
-		// Update or create the service
-
-		var sessionID string
-		if a := getAuthSession(ctx); a != nil {
-			sessionID = a.ID
-		}
-		if exists {
-			srv.Update(data)
-			if err := services.Update(ctx, tx, srv); err != nil {
-				return err
-			}
-			log.Debug(ctx, "postServiceRegisterHandler> update existing service %s(%d) registered for consumer %s", srv.Name, srv.ID, *srv.ConsumerID)
-		} else {
-			srv = &data
-			srv.ConsumerID = &consumer.ID
-
-			if err := services.Insert(ctx, tx, srv); err != nil {
-				return sdk.WithStack(err)
-			}
-			log.Debug(ctx, "postServiceRegisterHandler> insert new service %s(%d) registered for consumer %s", srv.Name, srv.ID, *srv.ConsumerID)
-		}
+		log.Debug(ctx, "postServiceRegisterHandler> update existing service %s(%d) registered for consumer %s", srv.Name, srv.ID, *srv.ConsumerID)
+	} else {
+		srv = data
+		srv.ConsumerID = &consumer.ID
 
-		if err := services.UpsertStatus(tx, *srv, sessionID); err != nil {
+		if err := services.Insert(ctx, tx, srv); err != nil {
 			return sdk.WithStack(err)
 		}
+		log.Debug(ctx, "postServiceRegisterHandler> insert new service %s(%d) registered for consumer %s", srv.Name, srv.ID, *srv.ConsumerID)
+	}
 
-		if len(srv.PublicKey) > 0 {
-			log.Debug(ctx, "postServiceRegisterHandler> service %s registered with public key: %s", srv.Name, string(srv.PublicKey))
-		}
+	var strRegion string
+	if srv.Region != nil {
+		strRegion = fmt.Sprintf(" from region %q", *srv.Region)
+	}
+	log.Info(ctx, "Registering service %s(%d) %s, consumer: %s, session %s", srv.Name, srv.ID, strRegion, consumer.ID, session.ID)
 
-		// For hatchery service we need to check if there are workers that are not attached to an existing hatchery
-		// If some worker's parent consumer match current hatchery consumer we will attach this worker to the new hatchery.
-		if srv.Type == sdk.TypeHatchery {
-			if err := worker.ReAttachAllToHatchery(ctx, tx, *srv); err != nil {
-				return err
-			}
-		}
+	_, err = authentication.LoadSessionByID(ctx, tx, session.ID)
```

Comment:
Why do we need this ? Maybe add a comment.

Reply:
it was a debugging step